ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -512,27 +542,60 @@ fn get_request_metadata(req: &Request<GetTransactionsRequest>) -> tonic::Result<
         .collect();
     let request_metadata: RequestMetadata =
         serde_json::from_str(&serde_json::to_string(&request_metadata_map).unwrap()).unwrap();
-    // TODO: update the request name if these are internal requests.
+    let request_metadata = RequestMetadata {
+        request_connection_id: Some(Uuid::new_v4().to_string()),
+        ..request_metadata
+    };
     Ok(request_metadata)
 }
 
 async fn channel_send_multiple_with_timeout(
     resp_items: Vec<TransactionsResponse>,
     tx: tokio::sync::mpsc::Sender<Result<TransactionsResponse, Status>>,
+    current_batch_start_time: Instant,
 ) -> Result<(), SendTimeoutError<Result<TransactionsResponse, Status>>> {
     for resp_item in resp_items {
-        let current_instant = std::time::Instant::now();
+        let response_size = resp_item.encoded_len();
+        let num_of_transactions = resp_item.transactions.len();
+        let start_version = resp_item.transactions.first().unwrap().version;
+        let end_version = resp_item.transactions.last().unwrap().version;
+        let start_version_txn_timestamp_in_sec = {
+            let timestamp = resp_item
+                .transactions
+                .first()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap();
+            timestamp.seconds as f64 + timestamp.nanos as f64 / 1_000_000_000.0
+        };
+        let end_version_txn_timestamp_in_sec = {
+            let timestamp = resp_item
+                .transactions
+                .last()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap();
+            timestamp.seconds as f64 + timestamp.nanos as f64 / 1_000_000_000.0
+        };
         tx.send_timeout(
             Result::<TransactionsResponse, Status>::Ok(resp_item),
             RESPONSE_CHANNEL_SEND_TIMEOUT,
         )
         .await?;
-        sample!(
-            SampleRate::Duration(Duration::from_secs(60)),
-            info!(
-                "[data service] response waiting time in seconds: {}",
-                current_instant.elapsed().as_secs_f64()
-            );
+        info!(
+            start_version = start_version,
+            end_version = end_version,
```

Comment:
maybe also have a metric for max version served?

Reply:
between connections or for current connection?

Reply:
For the current connection i suppose, since for between connections we can derive that with a `max()`. I'm mainly interested in seeing if we can answer any questions about QOS (e.g. will connections fight for egress bandwidth in the worst case) and also to help debug lagging connections

Reply:
We can get max version for a connection with the logs using the new `request_connection_id` added, and taking max of the ending_version. Then turn it into a metric later on. We're planning to have metrics for latest version at each step of the indexer.

Reply:
Wait end_verison is the max version served right? For consistency this is important. If we want to track end_version of the request we should call it start/end_version_from_request

Reply:
We originally agreed the fields should be normalized, but the event message defines which metric it is.

Reply:
For example
* message = "Request received". end_version would be the request end version
* message = "Data chunk received". end_version would be the max version of the data that's received.